### PR TITLE
Make compress plugin normalization of Accept-Encoding header compatible with normalization in core TS.

### PR DESCRIPTION
**Pull request by ywkaras**
Specifically, remove parameters on values, but if there is a valid q parameter which is equal to 0.0, remove the associated value from the value list.

### CODE REVIEW

**Comment by wkaras** [comment]
Don't review this yet.  I don't think it's safe for me to use strtof() unless in a nul-terminated string, I need to replace this.

**Comment by ywkaras** [comment]
Ready for review.

File: plugins/compress/misc.cc [hunk 1]
```diff
@@ -41,40 +42,69 @@ gzip_free(voidpf /* opaque ATS_UNUSED */, voidpf address)
...
+    ts::TextView param{value.take_prefix_at(';')};
+    ts::TextView name{param.take_prefix_at('=')};
+    name.trim(" \t");
+    if (strcasecmp("q", name) == 0) {
+      // If q value is valid and is zero, suppress compression types.
+      param.trim(" \t");
+      if (param) {
+        ts::TextView whole{param.take_prefix_at('.')};
+        whole.ltrim(" \t");
+        if ("0" == whole) {
```
Comment (bryancall) [inline plugins/compress/misc.cc]: So `q=0000`0 is would remove the encoding, but `q=0.0` wouldn't?

The code could be simplified by just looking at `q=0`.

Comment (ywkaras) [reply plugins/compress/misc.cc]: This is what the spec says about what is a valid q parameter:  https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.9

What I think I see in core TS is that, if the q parameter is malformed, act as thought it was not present at all:
https://github.com/apache/trafficserver/blob/c40271ae7543a87bc90b4278954b79d304bb70b0/proxy/http/HttpTransactCache.cc#L768
https://github.com/apache/trafficserver/blob/c40271ae7543a87bc90b4278954b79d304bb70b0/proxy/hdrs/HttpCompat.cc#L479

The core code is more permissive than mine, but that doesn't necessarily seem good.  My code is safer and slightly more performant I think.

**Comment by calavera** [comment]
I think this also solves a problem I bumped into this morning by chance. It looks like if you send this request, the compress plugin doesn't work because the normalization removes `br` when there is nothing else in the header:

```
curl -H 'Accept-Encoding: br' --compressed ...
```

I know it's probably a corner case, because nobody is going to ask only for brotli, but I was testing it, and I couldn't figure out why the plugin didn't work.

The loop iteration in this PR should fix that:

https://github.com/apache/trafficserver/pull/5917/files#diff-2a099dc0e13bc50ece3749c708a39e29R91

**Comment by calavera** [comment]
Actually, I think I bumped into a different issue, it looks like `br` is stripped from the Accept-Encoding header that's proxied to the origin :pensive:

File: plugins/compress/misc.cc [hunk 2]
```diff
@@ -41,40 +42,69 @@ gzip_free(voidpf /* opaque ATS_UNUSED */, voidpf address)
...
+      ts::TextView values(values_, val_len);
+      while (values) {
+        ts::TextView next{values.take_prefix_at(',')};
+        strip_ae_value(next);
+        if (strcasecmp("gzip", next) == 0) {
+          gzip = true;
+        } else if (strcasecmp("br", next) == 0) {
+          br = true;
+        } else if (strcasecmp("deflate", next) == 0) {
+          deflate = true;
```
Comment (ywkaras) [inline plugins/compress/misc.cc]: I should confess I snuck in an unrelated change here.  I don't see why we should suppress Brotli in favor of older algorithms since Brotii is new and improved.  No?

**Comment by zwoop** [comment]
Needed for other PRs.

Cherry-picked to v9.0.x branch.
